src/anonymousUserTracking/criteriaCompletionChecker.ts

```diff
@@ -323,9 +344,19 @@ class CriteriaCompletionChecker {
   }
 
   private doesItemMatchQueries(item: any, searchQueries: any[]): boolean {
-    const filteredSearchQueries = searchQueries.filter((searchQuery) =>
-      Object.keys(item).includes(searchQuery.field)
-    );
+    const filteredSearchQueries = [];
+    for (let i = 0; i < searchQueries.length; i++) {
```

Comment:
I know we need to add some lint rules to prevent some of these back and forths we have on PRs, but please use array methods over for loops when possible.

Reply:
Done